src/Principal.mo

```diff
@@ -21,10 +21,10 @@ module {
   /// Conversion.
   public func toText(p : Principal) : Text = debug_show(p);
 
-  private let anonymousPrincipal : Blob = "\04";
+  private let anonymousPrincipal : Principal = Prim.principalOfBlob "\04";
```

Comment:
The rhs ain't static, so the compiler will reject it as (conservatively) impure.

Reply:
Yeah, I gathered. Maybe sometime we'll get specialised literals for principals... Closing